[ntuple] DAOS object class customization (1/2): refactor `RDaosXxx` classes

jalopezg-r00t opened the pull request:
The first in the series of two pull-requests to allow users to customize the DAOS object class.

This PR applies the required changes to the `RDaosXxx` classes.  Also, libdaos_mock is extended to provide an implementation for the object class name to ID conversion functions.
Finally, unneeded `OC_xxx` constants were stripped from libdaos_mock/daos.h; this header should only contain declarations needed for testing.

Related to issue #8205.

jblomer reviewed (changes requested):
Nice improvement!

jalopezg-r00t commented on tree/ntuple/v7/inc/ROOT/libdaos_mock/daos.h:
```diff
@@ -106,245 +106,19 @@ int daos_event_fini(daos_event_t *ev);
-	OC_S4K,
-	OC_S8K,
+	OC_RP_XSF	= 80,
 	OC_SX,
```
Yes, thanks for spotting!  I kept the entire enumerator block.

jalopezg-r00t commented:
I made additional changes to one of the commits in the follow-up pull request.  As this was not merged yet, I also touched the corresponding part here.

@jblomer, could you please review commit 97c90bb5be7181443665be9b3ee44e82c9829f98?  Let me know if you have any suggestion.

jalopezg-r00t commented:
@phsft-bot build just on ROOT-debian-i386/cxx14

jalopezg-r00t commented:
@phsft-bot build just on ROOT-debian10-i386/cxx14

jblomer reviewed (changes requested):
I left a review on 6efaf8e

jalopezg-r00t commented:
> That's pretty dangerous. Who know how long class names will be in the future. The implementation just seems to use `strcpy`. I think that's something we can discuss with Johann -- if there is a DAOS defined limit that we can use. I'd very much prefer that. Lacking such a limit, I'd use a much larger one for ours, e.g. 64 or 128.

I completely agree, that's why I left a comment and asked you to re-review ;-)  I will ask Johann, or otherwise use a larger buffer.

jblomer reviewed (commented):
Could you pinpoint what's the change since the last review?  In general, I think it's easier if you add on commits rather than force push an updated history.

jalopezg-r00t commented:
> Could you pinpoint what's the change since the last review? In general, I think it's easier if you add on commits rather than force push an updated history.

I have updated the value (and documentation) of the `kOCNameMaxLength` constant.  Sorry, I squashed the commits as if the PR history was going to be merged, but I agree it is harder for you to review!  For subsequent PRs, I will add independent commits and will do the interactive rebase only after the review.  :-)

jalopezg-r00t commented:
@phsft-bot build just on windows10/cxx14
